inference-engine/src/legacy_api/src/convert_function_to_cnn_network.cpp

```diff
@@ -849,6 +849,97 @@ InferenceEngine::details::CNNLayerCreator::CNNLayerCreator(const std::shared_ptr
+            std::string value = transpose;
+            value.at(0) = ::toupper(value.at(0));
+            return value;
+        };
+
+        res->params["transpose_a"] = transpUppercaseFirstChar(params.at("transpose_a"));
+        res->params["transpose_b"] = transpUppercaseFirstChar(params.at("transpose_b"));
+        return res;
+    });
+
+    addSpecificCreator({"OneHotIE"}, [](const std::shared_ptr<::ngraph::Node>& node,
+                                      const std::map<std::string, std::string>& params) -> CNNLayerPtr {
+        LayerParams attrs = {node->get_friendly_name(), "OneHot", details::convertPrecision(node->get_output_element_type(0))};
+        auto res = std::make_shared<InferenceEngine::OneHotLayer>(attrs);
+        res->params = params;
+        return res;
+    });
+
+    addSpecificCreator({"PadIE"}, [](const std::shared_ptr<::ngraph::Node>& node,
+                                      const std::map<std::string, std::string>& params) -> CNNLayerPtr {
+        LayerParams attrs = {node->get_friendly_name(), "Pad", details::convertPrecision(node->get_output_element_type(0))};
+        auto res = std::make_shared<InferenceEngine::PadLayer>(attrs);
+
+        res->params["pad_mode"] = params.at("pad_mode");
+        res->params["pads_begin"] = params.at("pads_begin");
+        res->params["pads_end"] = params.at("pads_end");
+
+        if (params.at("pad_mode") == "constant") {
+            res->params["pad_value"] = params.at("pad_value");
+        }
+
+        return res;
+    });
+
+    addSpecificCreator({"FakeQuantize"}, [](const std::shared_ptr<::ngraph::Node>& node,
+                                                 const std::map<std::string, std::string>& params) -> CNNLayerPtr {
+        LayerParams attrs = {node->get_friendly_name(), "FakeQuantize", details::convertPrecision(node->get_output_element_type(0))};
+        auto res = std::make_shared<InferenceEngine::QuantizeLayer>(attrs);
+        res->params = params;
+        return res;
+    });
+
+    addSpecificCreator({"ConvolutionIE"}, [](const std::shared_ptr<::ngraph::Node>& node,
+                                                 const std::map<std::string, std::string>& params) -> CNNLayerPtr {
+        LayerParams attrs = {node->get_friendly_name(), "Convolution", details::convertPrecision(node->get_output_element_type(0))};
+        auto res = std::make_shared<InferenceEngine::ConvolutionLayer>(attrs);
+        res->params = params;
+
+        auto & rt_info = node->get_rt_info();
+        bool keep_constants(false);
+        if (auto attr = std::dynamic_pointer_cast<ngraph::VariantWrapper<int64_t>>(rt_info["keep_constants"])) {
+            keep_constants = attr->get();
+        }
+
+        // Restore output and kernel size
+        auto shape = node->get_input_shape(1);
+        shape.erase(shape.begin(), shape.begin() + 2);
+
+        res->params["kernel"] = Builder::asString(dynamic_cast<std::vector<size_t>&>(shape));
```

Comment:
This look like up casting so `static_cast` should be fine here.

Reply:
Corrected.